File: Arrays and Hashing/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> dict = {};
        for (int i=0; i<nums.size(); i++){
            if(auto search = dict.find(nums[i]); search == dict.end()){
                dict.insert(make_pair(target-nums[i], i));
            }
            else{
                return {search->second, i};
            }
        }
        return {};
    }
};

File: grind75/two_sum.py
#41ms
def twoSum(self, nums: List[int], target: int) -> List[int]:
        tally = {}
        for i in range (0, len(nums)):
            if nums[i] not in tally:
                tally[target-nums[i]] = i
            else:
                return [tally[nums[i]], i]
        return []